main.js

```diff
@@ -101,6 +92,47 @@ function displayNumbers(randomNumbers, currentScore) {
     });
 }
 
+function generatePositions(numbersToGenerate, randomNumbers) {
+    const table = [
+        [0, 1, 2, 3],
+        [4, 5, 6, 7],
+        [8, 9, 10, 11],
+        [12, 13, 14, 15]
+    ];
+    let positionValues = [];
+
+    document.querySelector('#play_area').innerHTML = '<div class="placeholder"> </div>'.repeat(15);
+
+    // using a for loop instead of forEach as the array.pop changes the length over each iteration
+    for (let i = 0; i < numbersToGenerate; i++) {
+        const possibleLocations = document.querySelectorAll('.placeholder:not(.numberBox)');
+        const position = Math.floor(Math.random() * possibleLocations.length);
+
+        possibleLocations[position].textContent = randomNumbers.pop();
+        possibleLocations[position].dataset.position = String(position);
+        possibleLocations[position].classList.add('numberBox');
+    }
+
+    document.querySelectorAll('.numberBox').forEach((numberBox) => {
+        positionValues.push(numberBox.dataset.position);
+    });
+
+
+    table.forEach((row) => {
+        let occurrences = 0;
+
+        positionValues.forEach((value) => {
+            if (row.includes(Number(value))) {
+                occurrences++;
+            }
+        });
+
+        if (occurrences > numbersToGenerate - 1) {
+            generatePositions(numbersToGenerate, randomNumbers);
```

Comment:
if they are all on 1 line, dont regenerate all of it, just move one of the numbers to a different line